### PR TITLE
Add .clang-format and docs

### PR DESCRIPTION
**Pull request by vagetablechicken**
#2272 
The problem of inconsistence style in Doris code is too big, it's hard to minimize modification when reformatting code.
So here, our aim is to make the style rules, tune the config in .clang-format.

Note: I choose clang-format-8.0+ to support richer  sytle options.

### CODE REVIEW

**Comment by kangkaisen** [comment]
Good job!

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    Priority:        3
+IndentCaseLabels: true
+IndentWidth:     4
+IndentWrappedFunctionNames: false
+KeepEmptyLinesAtTheStartOfBlocks: false
+MacroBlockBegin: ''
+MacroBlockEnd:   ''
+MaxEmptyLinesToKeep: 1
+NamespaceIndentation: None
+ObjCBlockIndentWidth: 2
```
Comment (imay) [inline .clang-format]: should be 4?

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+SpaceBeforeParens: ControlStatements
+SpaceInEmptyParentheses: false
+SpacesBeforeTrailingComments: 2
+SpacesInAngles:  false
+SpacesInContainerLiterals: true
+SpacesInCStyleCastParentheses: false
+SpacesInParentheses: false
+SpacesInSquareBrackets: false
+Standard:        Auto
+TabWidth:        8
```
Comment (imay) [inline .clang-format]: is this 4?

File: .clang-format (hunk 1 above)
Comment (vagetablechicken) [reply .clang-format]: The number of characters to use for indentation of ObjC blocks. It's irrelevant.

[IndentWidth:     4](https://github.com/apache/incubator-doris/pull/2724/files/921235ba8c28609f83ac591a4083b19cd5623a94#diff-69a68f7ce1ea3687a38d8e4705e6fa48R58) matters.

File: .clang-format (hunk 2 above)
Comment (vagetablechicken) [reply .clang-format]: We set `UseTab:          Never`,  so it's a dispensable option. It's ok, set 4 can reduce misunderstandings.

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**